### PR TITLE
Allow emitting build artifacts using "output" flag

### LINKED ISSUE #70: Allow to set build output to tar file
Would be nice (especially in combination with dryrun option) to set output to tar file where build result would be written to so that it could be checked with [trivy](https://aquasecurity.github.io/trivy) later on.

This could be done using command `--output=type=tar,dest=/path/to/tar/output`

### PR DESCRIPTION
**Pull request by djmaze**
closes #70

### CODE REVIEW

File: _docs/_index.md [hunk 1]
```diff
@@ -142,6 +142,9 @@ pull_image
 compress
 : enables compression og the build context using gzip
 
+output
+: sets output folder for build artifacts (format: `/drone/src/path`)
```
Comment (xoxys) [inline _docs/_index.md:146]: The syntax for the `output` flag seems to be a bit different according to https://docs.docker.com/engine/reference/commandline/buildx_build/#output

**Review by xoxys — CHANGES_REQUESTED** [review]
Thanks for your work. I'm not sure if we need to handle `--push` (https://github.com/thegeeklab/drone-docker-buildx/blob/17016c9f496dca2c3edaa66a8d1db5c6220b6381/plugin/docker.go#L87-L89) in case of `output` is set as this flag is just an `output` shorthand https://docs.docker.com/engine/reference/commandline/buildx_build/#push

File: _docs/_index.md (hunk 1 above)
Comment (djmaze) [reply _docs/_index.md:146]: I didn't think that any other usage that `local` would be useful. But well, people might want to export an image as well.

Also, I see that relative paths are allowed as well. So maybe we can change this to 

> (format: `path` or `type=TYPE[,KEY=VALUE]`)

instead?

Comment (xoxys) [reply _docs/_index.md:146]: For local trivy scans a tar export would be required as described in the issue. Making the helptext more generic sounds good.

**Comment by djmaze** [comment]
So, with this implementation, you need to specify `dry_run` explicitely if you use `output`. You mean we can rather skip adding the `--push` flag if `output` is set, right?

File: _docs/_index.md (hunk 1 above)
Comment (djmaze) [reply _docs/_index.md:146]: Oops, should've taken a closer look at the issue -_-

**Comment by xoxys** [comment]
Yeah I fear it might end up in some unwanted errors or results if --push and --output is specified. To avoid this I think we should always skip push if output is set.

File: plugin/docker.go [hunk 2]
```diff
@@ -84,7 +84,7 @@ func commandBuild(build Build, dryrun bool) *exec.Cmd {
 	}
 
 	args = append(args, build.Context)
-	if !dryrun && len(build.Tags.Value()) > 0 {
+	if !dryrun && build.Output == "" && len(build.Tags.Value()) > 0 {
```
Comment (xoxys) [inline plugin/docker.go:87]: I think this should be `if (!dryrun && len(build.Tags.Value()) > 0) || build.Output == ""`

**Review by xoxys — CHANGES_REQUESTED** [review]

File: plugin/docker.go (hunk 2 above)
Comment (xoxys) [reply plugin/docker.go:87]: I had a second look and I think your current condition is correct sorry.

**Review by xoxys — APPROVED** [review]

**Pull request merged**